lib/ui/recommendation.js

```diff
@@ -123,21 +123,25 @@ Recommendation.prototype = {
     this.events.publish('recommendation-mousemove', evt);
   },
   show: function(data) {
+    let payload = data.payload;
+    let ResultClass = data.resultClass;
```

Comment:
I don't think the recommendation-server should have knowledge of the UI classes. Better to have this recommendation class pick a UI class based on properties of the data returned by the server.